integration/tests/basic/erc/test_EIP1559.py

```diff
@@ -0,0 +1,1317 @@
+            accounts=self.accounts,
+            web3_client=self.web3_client,
+            access_list=None,
+        )
+
+    @pytest.mark.neon_only
+    def test_transfer_without_chain_id_negative(
+            self,
+            json_rpc_client: JsonRPCSession,
+    ):
+        sender = self.accounts[0]
+        recipient = self.web3_client.create_account()
+
+        tx_params = self.web3_client.make_raw_tx_eip_1559(
+            chain_id=None,
+            from_=sender.address,
+            to=recipient.address,
+            value=10000,
+            nonce="auto",
+            gas="auto",
+            max_priority_fee_per_gas="auto",
+            max_fee_per_gas="auto",
+            data=None,
+            access_list=None,
+        )
+
+        signed_tx = self.web3_client._web3.eth.account.sign_transaction(tx_params, sender.key)
+        response = json_rpc_client.send_rpc(method="eth_sendRawTransaction", params=signed_tx.rawTransaction.hex())
+        assert "result" not in response
+        assert "error" in response
+        assert response["error"]["code"] == -32000
+
+    def test_deploy_positive(self):
+        validate_deploy_positive(
+            accounts=self.accounts,
+            web3_client=self.web3_client,
+            access_list=None,
+        )
+
+    def test_contract_function_call_positive(
+            self,
+    ):
+        account = self.accounts[0]
+        contract_a, _ = self.web3_client.deploy_and_get_contract(
+            contract="EIPs/EIP161/contract_a_function.sol",
+            version="0.8.12",
+            account=account,
+            contract_name="ContractA",
+        )
+
+        data = contract_a.functions.deploy_contract().build_transaction()["data"]
+
+        tx_params = self.web3_client.make_raw_tx_eip_1559(
+            chain_id="auto",
+            from_=account.address,
+            to=contract_a.address,
+            value=None,
+            nonce="auto",
+            gas="auto",
+            max_priority_fee_per_gas="auto",
+            max_fee_per_gas="auto",
+            data=data,
+            access_list=None,
+        )
+
+        tx_receipt = self.web3_client.send_transaction(
+            account=account,
+            transaction=tx_params,
+        )
+
+        assert tx_receipt.type == 2
+        assert tx_receipt.status == 1, "ContractB was not deployed"
+        assert tx_receipt.logs, "ContractB was not deployed"
+        contract_b_address = tx_receipt.logs[0].address
+
+        contract_a_nonce = self.web3_client.get_nonce(contract_a.address)
+        assert contract_a_nonce == 2
+
+        contract_b: Contract = self.web3_client.get_deployed_contract(
+            address=contract_b_address,
+            contract_file="EIPs/EIP161/contract_b.sol",
+            contract_name="ContractB",
+        )
+
+        data = contract_b.functions.getOne().build_transaction()["data"]
+        tx_params = self.web3_client.make_raw_tx_eip_1559(
+            chain_id="auto",
```

Comment:
I purposely didn't assign any default values for parameters in `make_raw_tx_eip_1559`
the reasoning is this method will be used very seldom an if it does get used - I thought it might be a good idea for better readability
let me know if I should refactor it